buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/license/ProjectLicense.groovy

```diff
@@ -12,10 +12,69 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package com.google.firebase.gradle.plugins.license;
+package com.google.firebase.gradle.plugins.license
+
+import groovy.util.slurpersupport.GPathResult
 
 class ProjectLicense implements Serializable {
+    private static final FileNameFinder FINDER = new FileNameFinder()
+
     String name
-    List<URI> licenseUris
-    boolean isExplicit = true
+    List<URI> explicitLicenseUris
+    ProjectLicense parent
+
+    List<URI> getLicenseUris() {
+        return getUris(this)
+    }
+
+    private static List<URI> getUris(ProjectLicense license) {
+        if (license == null) {
+            return []
+        }
+        return license.explicitLicenseUris ?: getUris(license.parent)
+    }
+
+    static ProjectLicense inferProjectLicenseFromArtifact(File artifact) {
+        return parsePom(findPomRelativeToArtifact(artifact))
+    }
+
+    /**
+     * According to gradle's dependency cache layout(files-2.1), the artifacts are stored in the following way:
+     *
+     * <ul>
+     *   <li>${groupId}/${artifactId}/${version}/sha1(file)/*.{jar|aar}
+     *   <li>${groupId}/${artifactId}/${version}/sha1(file)/*.pom
+     * <ul>
+     *
+     * <p>So to get to a pom file based on the aar, we need to look for it based on the following
+     * pattern: `../{@literal *}/*.pom`
+     */
+    private static File findPomRelativeToArtifact(File artifact) {
+        return FINDER.getFileNames(artifact.parentFile.parentFile.absolutePath, '*/*.pom').collect {
+            new File(it)
+        }.find()
+    }
+
+    private static ProjectLicense parsePom(File pomFile) {
+        if (pomFile == null) return null
+        pomFile.withReader {
+            def project = new XmlSlurper().parse(it)
+            def name = project.name
+            def licenses = project.licenses.license.collect { URI.create(it.url as String) }
+            return new ProjectLicense(
+                    name: name,
+                    explicitLicenseUris: licenses,
+                    parent: determineParent(pomFile.parentFile.parentFile.parentFile.parentFile.parentFile, project.parent))
+        }
+    }
+
+    private static ProjectLicense determineParent(File baseDir, GPathResult parentNode) {
```

Comment:
Worth leaving a comment on why we need to look up licenses from the parent for some libs